evergreen/evergreen.go

```diff
@@ -0,0 +1,208 @@
+package evergreen
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-wordwrap"
+	"gopkg.in/yaml.v3"
+)
+
+type Config struct {
+	Variants []Variant `yaml:"buildvariants"`
+}
+
+type Variant struct {
+	Name        string `yaml:"name"`
+	DisplayName string `yaml:"display_name"`
+	Tasks       []Task `yaml:"tasks"`
+}
+
+type Task struct {
+	Name string `yaml:"name"`
+}
+
+func Load() (*Config, error) {
+	_, pkgPath, _, _ := runtime.Caller(0)
+	common, err := ioutil.ReadFile(filepath.Join(filepath.Dir(pkgPath), "..", "common.yml"))
+	if err != nil {
+		return nil, err
+	}
+
+	var config Config
+	err = yaml.Unmarshal(common, &config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &config, nil
+}
+
+func (v *Variant) includesLatest() bool {
+	for _, t := range v.Tasks {
+		if t.Name == ".latest" {
+			return true
+		}
+	}
+	return false
+}
+
+var versionRE = regexp.MustCompile(`\.(\d+)\.(\d+)$`)
+
+func (v *Variant) mostRecentServerVersion() (string, error) {
+	var versions [][2]int
+	for _, t := range v.Tasks {
+		if matches := versionRE.FindStringSubmatch(t.Name); len(matches) != 0 {
+			maj, err := strconv.Atoi(matches[1])
+			if err != nil {
+				return "", err
+			}
+			min, err := strconv.Atoi(matches[2])
+			if err != nil {
+				return "", err
+			}
+			versions = append(versions, [2]int{maj, min})
+		}
+	}
+	if len(versions) == 0 {
+		return "", nil
+	}
+
+	sort.SliceStable(versions, func(i, j int) bool {
+		majI := versions[i][0]
+		majJ := versions[j][0]
+		minI := versions[i][1]
+		minJ := versions[j][1]
+
+		if majI != majJ {
+			return majI < majJ
+		}
+		return minI < minJ
+	})
+
+	return fmt.Sprintf("%d.%d", versions[0][0], versions[0][1]), nil
+}
+
+type alias struct {
+	comment string
+	variant string
+	tasks   string
+}
+
+var lineStartRE = regexp.MustCompile(`(?m)^`)
+
+func (c *Config) GitHubPRAliasesYAML() (string, error) {
+	aliases := []alias{
+		{
+			comment: "Run all static analysis tasks.",
+			variant: `^static$`,
+			tasks:   `.*`,
+		},
+		{
+			comment: "Run unit, sign, and push for every platform.",
+			variant: `.*`,
+			tasks:   `^(unit|sign|push)$`,
```

Comment:
I think we should not do sign or push for this. I don't think we get much out of it. Push gets skipped on patches anyway.